### PR TITLE
Kafka consumer

### PR DESCRIPTION
**Pull request by sectioneight**
OSS version of the kafka consumer.

A lot of the code was great as-is, just needed some tweaking to remove Uber specific stuff, and add errchecks.

GFM-65

### CODE REVIEW

**Comment by sectioneight** [comment]
@glibsm @akshayjshah the old PR was closed out in a different repo, re-opened here after cherry-picking.

File: modules/kafka/consumer/config_test.go [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+	"time"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestConfigValidate(t *testing.T) {
+	valid := fakeConfig()
+
+	hasHostFile := fakeConfig()
+	hasHostFile.HostFile = "/etc/uber/kafka8/clusters.yaml"
```
Comment (anuptalwalkar) [inline modules/kafka/consumer/config_test.go]: We should have some tmp path here and not point to etc/uber

**Comment by sectioneight** [comment]
So, what's the next step here @akshayjshah @prashantv? Should we create a new uber-go repo for kafka consumer? Would you like to be the owner of that or us? Or should we just try to merge this in uberfx?

**Comment by sectioneight** [comment]
Closing until we figure out the right home for this. Meeting set up for next week.

**Pull request closed without merging**